platform/mbed_assert.h

```diff
@@ -49,6 +49,65 @@ do {                                                     \
 } while (0)
 #endif
 
+
+#define _MBED_CONCAT_(a, b) a##b
```

Comment:
I'm just wondering if these kind of macro can be packed together. Preprocessor token concatenation is a common operation and in many places across the code base it is not done correctly (arguments not expanded...).

This might be done in another PR.